ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -295,6 +300,26 @@ private Response executeStatements(
     }
   }
 
+  private void waitForPreviousDistributedStatementToBeHandled(final KsqlEntityList entities) {
+    final ArrayList<KsqlEntity> reversed = new ArrayList<>(entities);
+    Collections.reverse(reversed);
+
+    reversed.stream()
+        .filter(e -> e instanceof CommandStatusEntity)
+        .map(cs -> ((CommandStatusEntity) cs).getCommandSequenceNumber())
+        .filter(Objects::nonNull)
```

Comment:
You should be able to change the return type of `getCommandSequenceNumber#getCommandSequenceNumber()` from `Long` to `long`, and remove the need for this filter a result.

Reply:
Thanks. Done.